### PR TITLE
Remove defunct properties from docs

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Requested by @electrum in https://trinodb.slack.com/archives/CGB0QHWSW/p1716587057716529
and expanded to remove all defunct properties from GlueHiveMetastoreConfig

## Additional context and related issues

v1 class still supports some but that is not documented and probably not desired to be documented at this stage either.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 1]
```diff
@@ -511,7 +494,7 @@ iceberg.catalog.type=rest
 iceberg.rest-catalog.uri=http://iceberg-with-rest:8181
 ```
 
-The REST catalog supports [view management](sql-view-management) 
+The REST catalog supports [view management](sql-view-management)
```
Comment (martint) [inline docs/src/main/sphinx/object-storage/metastores.md:497]: Unrelated change

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ are also available. They are discussed later in this topic.
     metastore calls for that partition.
   - `false`
 * - `hive.metastore-cache.cache-missing-stats`
-  - Enable caching the fact that table statistics for a specific table are 
+  - Enable caching the fact that table statistics for a specific table are
```
Comment (martint) [inline docs/src/main/sphinx/object-storage/metastores.md:63]: Unrelated change

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md]: undone

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md]: undone

**Review by martint — APPROVED** [review]
